File: sdk/go/authpolicy.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
)

type AuthPolicySettings struct {
	TenantID                string   `json:"tenant_id"`
	PasswordMinLength       int      `json:"password_min_length,omitempty"`
	PasswordRequireUppercase bool    `json:"password_require_uppercase,omitempty"`
	PasswordRequireNumbers  bool     `json:"password_require_numbers,omitempty"`
	PasswordRequireSymbols  bool     `json:"password_require_symbols,omitempty"`
	AllowedLoginMethods     []string `json:"allowed_login_methods,omitempty"`
	MfaRequired             bool     `json:"mfa_required,omitempty"`
	SessionDurationSeconds  int      `json:"session_duration_seconds,omitempty"`
	UpdatedAt               string   `json:"updated_at,omitempty"`
}

type UpdateAuthPolicyInput struct {
	PasswordMinLength        *int     `json:"password_min_length,omitempty"`
	PasswordRequireUppercase *bool    `json:"password_require_uppercase,omitempty"`
	PasswordRequireNumbers   *bool    `json:"password_require_numbers,omitempty"`
	PasswordRequireSymbols   *bool    `json:"password_require_symbols,omitempty"`
	AllowedLoginMethods      []string `json:"allowed_login_methods,omitempty"`
	MfaRequired              *bool    `json:"mfa_required,omitempty"`
	SessionDurationSeconds   *int     `json:"session_duration_seconds,omitempty"`
}

type AuthPolicy struct{ http *httpClient }

func (r *AuthPolicy) Get(ctx context.Context, tenantID string) (*AuthPolicySettings, error) {
	var out AuthPolicySettings
	err := r.http.do(ctx, http.MethodGet, "/v1/tenants/"+url.PathEscape(tenantID)+"/auth-policy", nil, nil, &out, true)
	return &out, err
}

func (r *AuthPolicy) Update(ctx context.Context, tenantID string, in UpdateAuthPolicyInput) (*AuthPolicySettings, error) {
	var out AuthPolicySettings
	err := r.http.do(ctx, http.MethodPut, "/v1/tenants/"+url.PathEscape(tenantID)+"/auth-policy", nil, in, &out, false)
	return &out, err
}

File: sdk/js/node/src/authPolicy.ts
import type { HttpClient } from "./client.js";

export interface AuthPolicySettings {
  tenant_id: string;
  password_min_length?: number;
  password_require_uppercase?: boolean;
  password_require_numbers?: boolean;
  password_require_symbols?: boolean;
  allowed_login_methods?: string[];
  mfa_required?: boolean;
  session_duration_seconds?: number;
  updated_at?: string;
}

export interface UpdateAuthPolicyInput {
  password_min_length?: number;
  password_require_uppercase?: boolean;
  password_require_numbers?: boolean;
  password_require_symbols?: boolean;
  allowed_login_methods?: string[];
  mfa_required?: boolean;
  session_duration_seconds?: number;
}

export class AuthPolicy {
  constructor(private readonly http: HttpClient) {}

  get(tenantId: string): Promise<AuthPolicySettings> {
    return this.http.get<AuthPolicySettings>(
      `/v1/tenants/${encodeURIComponent(tenantId)}/auth-policy`,
    );
  }

  update(tenantId: string, input: UpdateAuthPolicyInput): Promise<AuthPolicySettings> {
    return this.http.request<AuthPolicySettings>(
      "PUT",
      `/v1/tenants/${encodeURIComponent(tenantId)}/auth-policy`,
      { body: input },
    );
  }
}
